scripts/roomba_controller.py

```diff
@@ -0,0 +1,167 @@
+#! /usr/bin/env python
+
+import sys
+import rospy
+from iarc7_msgs.msg import OdometryArray
+import actionlib
+import tf2_ros
+import threading
+from iarc7_motion.msg import QuadMoveGoal, QuadMoveAction
+from iarc7_safety.SafetyClient import SafetyClient
+
+class RoombaRequests(object):
+    def __init__(self, frame_id, tracking_mode, time_to_hold_once_done):
+        self.frame_id = frame_id
+        self.tracking_mode = tracking_mode
+        self.time_to_hold = time_to_hold_once_done
+
+class RoombaControllerStates(object):
+    tracking = 1
+    block = 2
+    hit = 3
+    recover_from_success = 4
+    hold_position = 5
+    completed = 6
+    failed_task = 7
+    failed_recovery = 8
+    invalid_state = 9
+
+class RoombaController(object): 
+
+    def __init__(self, roomba_request, client, status_callback):
+        self._status_callback = status_callback
+        self._client = client
+        roomba_id = roomba_request.frame_id
+        self._roomba_id = roomba_id [0:len(roomba_id)-10]
```

Comment:
No space before the `[` here, might want spaces around the `:` but that's optional

Reply:
Also you need a comment explaining what the `10` is

Reply:
fixed, but keeping : in because it looks cool